src/Bicep.Cli/Commands/PublishTypeCommand.cs

```diff
@@ -0,0 +1,146 @@
+using System.IO.Abstractions;
+using System.Threading.Tasks;
+using Azure.Bicep.Types;
+using Azure.Bicep.Types.Index;
+using Azure.Bicep.Types.Serialization;
+using Bicep.Cli.Arguments;
+using Bicep.Cli.Logging;
+using Bicep.Cli.Services;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Exceptions;
+using Bicep.Core.Features;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Registry;
+using Bicep.Core.Resources;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.Workspaces;
+using Microsoft.Extensions.Logging;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+using Bicep.Core.TypeSystem.Providers.Az;
+using Bicep.Core.TypeSystem.Az;
+using Bicep.Core.UnitTests;
+using Bicep.Core.TypeSystem.Providers;
+
+namespace Bicep.Cli.Commands
+{
+    public class PublishTypeCommand : ICommand
+    {
+        private readonly IDiagnosticLogger diagnosticLogger;
+        private readonly CompilationService compilationService;
+        private readonly CompilationWriter compilationWriter;
+        private readonly IModuleDispatcher moduleDispatcher;
+        private readonly IFileSystem fileSystem;
+        private readonly IFeatureProviderFactory featureProviderFactory;
+        private readonly IOContext ioContext;
+        private readonly ILogger logger;
+
+        public PublishTypeCommand(
+            IDiagnosticLogger diagnosticLogger,
+            CompilationService compilationService,
+            IOContext ioContext,
+            ILogger logger,
+            CompilationWriter compilationWriter,
+            IModuleDispatcher moduleDispatcher,
+            IFileSystem fileSystem,
+            IFeatureProviderFactory featureProviderFactory)
+        {
+            this.diagnosticLogger = diagnosticLogger;
+            this.compilationService = compilationService;
+            this.compilationWriter = compilationWriter;
+            this.moduleDispatcher = moduleDispatcher;
+            this.fileSystem = fileSystem;
+            this.featureProviderFactory = featureProviderFactory;
+            this.ioContext = ioContext;
+            this.logger = logger;
+        }
+
+        [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "<Pending>")]
+        public async Task<int> RunAsync(PublishTypeArguments args)
+        {
+            var indexPath = PathHelper.ResolvePath(args.IndexFile);
+            var inputUri = PathHelper.FilePathToFileUrl(indexPath);
+            var features = featureProviderFactory.GetFeatureProvider(PathHelper.FilePathToFileUrl(indexPath));
+            var documentationUri = args.DocumentationUri;
+            var typeReference = ValidateReference(args.TargetTypeReference, inputUri);
+            var overwriteIfExists = args.Force;
+
+            //attempt to validate types here, move to better location later
```

Comment:
I think it might be worth putting an experimental warning here, just in case people do discover the capability.

See https://github.com/Azure/bicep/blob/7e89b5a02e531a95803115097ce025d55564076d/src/Bicep.Cli/Commands/JsonRpcCommand.cs#L33 for an example